flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/FlinkBatchRuleSets.scala

```diff
@@ -307,6 +307,9 @@ object FlinkBatchRuleSets {
     CoreRules.PROJECT_TO_CALC,
     FlinkCalcMergeRule.INSTANCE,
 
+    // remove output of rank number when it is not used by successor calc
+    RedundantRankNumberColumnRemoveRule.INSTANCE,
```

Comment:
please move this rule close to other rank rules

Reply:
The rule should be in a HepPlanner phase instead of VolcanoPlanner phase.